src/components/fields/contact-field/contact-field.tsx

```diff
@@ -36,8 +38,21 @@ export const ContactField = (props: IGenericFieldProps<IContactFieldSchema>) =>
 	useEffect(() => {
 		const contactNumberRule = validation?.find((rule) => "contactNumber" in rule);
 		const singaporeRule = contactNumberRule?.["contactNumber"]?.["singaporeNumber"];
+		const internationalNumberRule = contactNumberRule?.["contactNumber"]?.["internationalNumber"];
 		const errorMessage = contactNumberRule?.["errorMessage"];
 
+		if (internationalCodeMap.has(internationalNumberRule)) {
+			setCountryValue(internationalNumberRule);
+			setFixedCountry(true);
+		} else if ((["default", "house", "mobile"] as TSingaporeNumberRule[]).includes(singaporeRule)) {
+			setCountryValue("Singapore");
+			setFixedCountry(true);
+		}
+
+		if (!fixedCountry) {
```

Comment:
state updates run asynchronously, any changes to `fixedCountry` will not be updated in time for this check so you should avoid checking against the state directly